source/dub/git.d

```diff
@@ -0,0 +1,40 @@
+// functionality to supply packages from git submodules
+module dub.git;
+
+import dub.dependency;
+import dub.internal.vibecompat.core.file;
+import dub.package_;
+import dub.packagemanager;
+
+import std.algorithm;
+import std.exception : enforce;
+import std.range;
+import std.string;
+
+/// adds the git submodules checked out in the root path as direct packages
+public void addGitSubmodules(PackageManager packageManager, NativePath rootPath) {
+	import std.process : execute;
+
+	auto rootScmPath = rootPath ~ ".git";
+	const submoduleInfo = execute([
+		"git",
+		"-C", rootPath.toNativeString,
+		"--git-dir=" ~ (rootScmPath.relativeTo(rootPath)).toNativeString,
+		"submodule", "status"]);
+
+	enforce(submoduleInfo.status == 0,
+		format("git submodule status exited with error code %s: %s", submoduleInfo.status, submoduleInfo.output));
+
+	foreach (line; submoduleInfo.output.lines) {
+		const parts = line.split(" ").map!strip.filter!(a => !a.empty).array;
+		const subPath = rootPath ~ parts[1];
+		const packageFile = Package.findPackageFile(subPath);
+
+		if (!packageFile.empty) {
+			const scmPath = rootPath ~ NativePath(".git/modules/" ~ parts[1]);
+			packageManager.getOrLoadPackage(subPath, packageFile, false, scmPath);
+		}
+	}
+}
+
+private alias lines = text => text.split("\n").map!strip.filter!(a => !a.empty);
```

Comment:
Can we at least split on https://dlang.org/phobos/std_ascii.html#newline?

Reply:
done